lib/srv/app/aws/handler_test.go

```diff
@@ -157,6 +159,50 @@ func TestAWSSignerHandler(t *testing.T) {
 	}
 }
 
+func TestURLForResolvedEndpoint(t *testing.T) {
+	tests := []struct {
+		name                 string
+		inputReq             *http.Request
+		inputResolvedEnpoint *endpoints.ResolvedEndpoint
+		requireError         require.ErrorAssertionFunc
+		expectURL            string
+	}{
+		{
+			name:     "bad resolved endpoint",
+			inputReq: mustNewRequest(t, "GET", "http://1.2.3.4/hello/world?aa=2", nil),
+			inputResolvedEnpoint: &endpoints.ResolvedEndpoint{
+				URL: string([]byte{0x05}),
+			},
+			requireError: require.Error,
+		},
+		{
+			name:     "replaced host and scheme",
+			inputReq: mustNewRequest(t, "GET", "http://1.2.3.4/hello/world?aa=2", nil),
+			inputResolvedEnpoint: &endpoints.ResolvedEndpoint{
+				URL: "https://local.test.com",
+			},
+			expectURL:    "https://local.test.com/hello/world?aa=2",
+			requireError: require.NoError,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			actualURL, err := urlForResolvedEndpoint(test.inputReq, test.inputResolvedEnpoint)
+			require.Equal(t, test.expectURL, actualURL)
+			test.requireError(t, err)
+		})
+	}
+}
+
+func mustNewRequest(t testing.TB, method, url string, body io.Reader) *http.Request {
```

Comment:
nit: Just for consistency - at first glance it might be a bit confused why  TB interface is used instead of where as all function in this file operates on `*testing.T` object. 
```suggestion
func mustNewRequest(t testing.T, method, url string, body io.Reader) *http.Request {
```

Reply:
I can revert but i really think we should start doing `testing.TB` more often. It took me a really long time to write a `BenchmarkXXX` (for a different issue) to convert all our suite setup functions to work with `testing.B`.